### PR TITLE
[BugFix] Fix the failure of querying extenal table that contains struct columns

### LINKED ISSUE #9020: Select from external hive table failed when original table contains map and struct
### Steps to reproduce the behavior (Required)
- create hive table:
```
CREATE EXTERNAL TABLE `surface_tracking_enriched_events`(`rheosheader` struct<eventcreatetimestamp:bigint,eventsenttimestamp:bigint,schemaid:int,eventid:string,producerid:string> COMMENT '', `datasrc` string COMMENT '', `bizdtl` map<string,string> COMMENT '', `trackableid` string COMMENT '', `itemdtl` map<string,string> COMMENT '')
PARTITIONED BY (`session_start_dt` string)
ROW FORMAT SERDE 'org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe'
WITH SERDEPROPERTIES (
  'serialization.format' = '1'
)
STORED AS
  INPUTFORMAT 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat'
  OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat'
TBLPROPERTIES (
  'PARQUET.COMPRESS' = 'SNAPPY'
)
```
- insert data:
```
INSERT INTO surface_tracking_enriched_events PARTITION(session_start_dt="2022-07-15") select NAMED_STRUCT('eventcreatetimestamp',100L,'eventsenttimestamp',200L,'schemaid',300,'eventid','EVENTID1','producerid','PRODUCERID') as rheosheader, "datasrc", map("Mathematics","78"), "trackableid", map("Mathematics","78");
```
- create external table:
```
CREATE EXTERNAL TABLE `surface_tracking_enriched_events_external_v2` (
  `trackableid` varchar(65533) NULL COMMENT "",
  `session_start_dt` varchar(65533) NULL COMMENT ""
) ENGINE=HIVE 
COMMENT "PARTITION BY (session_start_dt)"
PROPERTIES (
"database" = "myhive",
"table" = "surface_tracking_enriched_events",
"resource" = "hive0"
);
```

- select from external table

### Expected behavior (Required)
return result

### Real behavior (Required)
```
MySQL [myhive]> select * from surface_tracking_enriched_events_external_v2;
ERROR 1064 (HY000): failed to find shared buffer based on offset
```

### StarRocks version (Required)
main branch

### PR DESCRIPTION
**Pull request by GavinMar**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9020

## Problem Summary(Required) ：

This PR is mainly used to solve two problems:

1. For map or struct columns, parquet reader parses the column meta to a flat vector,
each column meta should be located by `physical_column_index` rather than `col_idx_in_parquet`.
The current code uses `col_idx_in_parquet` when setting io ranges, while uses `physical_column_index` when reading column data, resulting in incorrect column meta information and mismatched data offset.

2.  For parquet file that contains some upper case column names, the real data may not be read correctly for the mismatched column names, because we convert all column to lower case before reading parquet files.

### CODE REVIEW

**Review by imay — DISMISSED** [review]
LGTM, I think it is better to have unit test cover

**Comment by GavinMar** [comment]
>

**Comment by GavinMar** [comment]
> LGTM, I think it is better to have unit test cover

Yes, unit test for parquet is indeed insufficient, I'll add some more later.

File: be/src/formats/parquet/schema.cpp [hunk 1]
```diff
@@ -356,7 +356,7 @@ std::string SchemaDescriptor::debug_string() const {
 
 int SchemaDescriptor::get_column_index(const std::string& column) const {
     for (size_t i = 0; i < _fields.size(); i++) {
-        if (_fields[i].name == column) {
+        if (strcasecmp(_fields[i].name.c_str(), column.c_str()) == 0) {
```
Comment (DorianZheng) [inline be/src/formats/parquet/schema.cpp:359]: Could you take a look if ORC share same issue?

**Comment by GavinMar** [comment]
Another PR  #9420 has been exsit.

**Pull request closed without merging**